### PR TITLE
Updated MFENCE.VMA description

### PR DESCRIPTION
**Pull request by eckhard-delfs-qualcomm**
As suggested by ved in https://github.com/riscv/riscv-smmtt/issues/72

### CODE REVIEW

File: chapter3.adoc [hunk 1]
```diff
@@ -188,8 +188,21 @@ respective sections in this specification.
 The `MFENCE.SPA` fence instruction is used to synchronize updates to supervisor
 domain access-permissions with current execution.
 `MFENCE.SPA` is only valid in M-mode. If operand rs1≠x0, it
-specifies a single physical address, and if rs2≠x0, it specifies
-a single SDID. Executing a `MFENCE.SPA` guarantees that any previous stores
+specifies a single physical address, and if rs2≠x0, it specifies a single SDID.
```
Comment (SiFiveHolland) [inline chapter3.adoc:191]: What is the granularity of an `MFENCE.SPA` with an address? This wording suggests that the granularity is a single byte, which seems rather unlikely. Compare the description of `SFENCE.VMA` in the privileged spec (emphasis mine):

> If rs1≠x0 and rs2=x0, the fence orders only reads and writes made to **leaf page table entries corresponding to the virtual address** in rs1, for all address spaces.

Should `MFENCE.SPA` do the same? In this context, if a MTTL2 entry has a type other than MTT_L1_DIR, would a "leaf page table entry" refer to the 32 MiB covered by that single MTTL2 entry, or the 2 MiB/4 MiB/1 GiB page containing the physical address?

Comment (eckhard-delfs-qualcomm) [reply chapter3.adoc:191]: Correct... we could add this statement for the cases where rs1≠x0:

_If rs1 corresponds to an `MTT_L1_DIR` mapping, the fence applies to a 4 KiB page. If rs1 corresponds to a `2M_PAGES` or `4M_PAGES` mapping, the fence applies to a 2 MiB or 4 MiB range. If rs1 corresponds to one of the `1G_*` mappings, the fence applies to a 1 GiB range._

Comment (rsahita) [reply chapter3.adoc:191]: sgtm

Comment (rsahita) [reply chapter3.adoc:191]: @eckhard-delfs-qualcomm please update - thanks

Comment (rsahita) [inline chapter3.adoc:191]:
```suggestion
specifies a single physical address, and if rs2≠x0, it specifies a single SDID. 

If rs1 corresponds to an MTT_L1_DIR mapping, the fence applies to
a 4 KiB page. If rs1 corresponds to a 2M_PAGES or 4M_PAGES mapping,
the fence applies to a 2 MiB or 4 MiB range. If rs1 corresponds to one
of the 1G_* mappings, the fence applies to a 1 GiB range.

```

**Review by rsahita — APPROVED** [review]

**Comment by rsahita** [comment]
Merging this PR given no other open comments.

**Pull request merged**